derp/derphttp/derphttp_client.go

```diff
@@ -757,6 +757,9 @@ func (c *Client) dialNode(ctx context.Context, n *tailcfg.DERPNode) (net.Conn, e
 			}
 			dst := cmp.Or(dstPrimary, n.HostName)
 			port := "443"
+			if !c.useHTTPS() {
```

Comment:
Drive-by: when testing locally, using a `-dev` mode derper did not work we didn't implement the logic that [this](https://github.com/tailscale/tailscale/pull/14122/files#diff-99911aec55c961ef16b36e90b0f5790bae75921c381504b4f7f09f9ea7c1aa09L245) comment says we have.